### ISSUE TITLE
Hope I'm not flamed for not being a coder - but.. 

### ISSUE THREAD

**Issue by brianr-dk** [collectd/collectd#3219]
I simply cannot get "WsrepStats true" to work. 

It fails with 
"Jul 15 10:50:43 ubuntuserver-1 collectd[13700]: mysql plugin: Option `wsrepstats' not allowed here."

I'm running Ubuntu 16.04 and Collectd 5.5.1 
The database is MariaDB 10.2, but that seems irelevant, since it *can* connect and collect mysql and innodbstats just fine. 
I've tried placing "wsrepstats" inside or outside the database section/block, but nothing makes it happy.   The C-code doesn't shed any light for me, sorry... 

https://github.com/collectd/collectd/blob/master/src/mysql.c 

Here's my collectd.conf: 

<Plugin mysql>
  <Database "galera_ubuntuserver-1">
    Host "localhost"
    User "collectd"
    Password "Fat4ohse"
    Port "3306"
        wsrepstats      true
        InnodbStats true
        #    MasterStats true
  </Database>
  #       WsrepStats true
  #       InnodbStats true

</Plugin>
 

## Expected behavior
That plugin:myslq starts pushing wsrep metrics collected from mysql/mariaDB into my Grafana/Graphite remote host (other mysql counters and innodb works splendid in Grafana ) ;) 

## Actual behavior
" WsrepStats not allowed here. " it shouts at me.

## Steps to reproduce
Installl collectd, 
Fiddle with the installed example config file, and see that wsrep is not mentioned, only master/slave. 
Happily discover through google that someone actually thought of Galera/wsrep counters and already included them as an "option" - even if those options are not mentioned in the file installed (from 2012?).

So close to a great take-off - but I'm apparently too inept to make it fly :/ 

Thanks in advance for any hints to the cause of my failure.

**Comment by rpv-tomsk** [comment]
Do you understand what is "wsrep" is about?

Do you can confirm, your MariaDB really has "wsrep" metrics/variables?

**Comment by rpv-tomsk** [comment]
>Here's my collectd.conf:

```
Host "localhost" User "collectd" Password "Fat4ohse" Port "3306" wsrepstats true InnodbStats true # MasterStats true # WsrepStats true # InnodbStats true
```

Your collectd.conf has incorrect formatting.

**Comment by brianr-dk** [comment]
Thanks rpv-tomsk 
Yes I know wsrep metrics, I'm running a 6-node Galeracluster on Mariadb and was so happy when I saw this option is possible in Collectd. 
But I can't make it work here, not even with your config, it won't accept the WsrepStats. 
(I've actually tried entering wrong option like "wsrepWrongName true" and the same happens, collectd/plugin:MySql just reports that "wsrepWrongName" is not allowed. How do one check if the supplied version of the plugin:MySql actually knows of that parameter/option werepstats ? )

I take it that you have the config file like this: 
<Plugin mysql> 
<database "SomeName>
    Your config lines here
</database>
</plugin> 
??

Do you also have collectd 5.5.1  (IS this wsrepstats possible in 5.5.1?)  


Without the <Database "Name">  arguments in my configfile, it fails with: 
  Jul 15 13:15:15 ubuntuserver-1 collectd[14054]: mysql plugin: Option "Host" not allowed here.
  Jul 15 13:15:15 ubuntuserver-1 collectd[14054]: mysql plugin: Option "User" not allowed here.
  Jul 15 13:15:15 ubuntuserver-1 collectd[14054]: mysql plugin: Option "Password" not allowed here.
  Jul 15 13:15:15 ubuntuserver-1 collectd[14054]: mysql plugin: Option "Port" not allowed here.
  Jul 15 13:15:15 ubuntuserver-1 collectd[14054]: mysql plugin: Option "wsrepstats" not allowed here.
  Jul 15 13:15:15 ubuntuserver-1 collectd[14054]: mysql plugin: Option "InnodbStats" not allowed here.
  Jul 15 13:15:15 ubuntuserver-1 collectd[14058]: mysql plugin: Option "Host" not allowed here.
  Jul 15 13:15:15 ubuntuserver-1 collectd[14058]: mysql plugin: Option "User" not allowed here.
  Jul 15 13:15:15 ubuntuserver-1 collectd[14058]: mysql plugin: Option "Password" not allowed here.
  Jul 15 13:15:15 ubuntuserver-1 collectd[14058]: mysql plugin: Option "Port" not allowed here.
  Jul 15 13:15:15 ubuntuserver-1 collectd[14058]: mysql plugin: Option "wsrepstats" not allowed here.
  Jul 15 13:15:15 ubuntuserver-1 collectd[14058]: mysql plugin: Option "InnodbStats" not allowed here.
  Jul 15 13:15:15 ubuntuserver-1 collectd[14058]: supervised by systemd, will signal readyness
  Jul 15 13:15:15 ubuntuserver-1 collectd[14058]: Initialization complete, entering read-loop.

And WITH the  <Database "Name">  in my config , so the plugin:MySql  looks like this: 
<Plugin mysql>
        <Database "Galera">
        Host "localhost"
        User "collectd"
        Password "Fat4ohse"
        Port "3306"
        wsrepstats true
        InnodbStats true
        </Database>
</Plugin>

It fails like this (same message twice?) : 
  Jul 15 13:16:16 ubuntuserver-1 collectd[14079]: mysql plugin: Option `wsrepstats' not allowed here.
  Jul 15 13:16:16 ubuntuserver-1 collectd[14082]: mysql plugin: Option `wsrepstats' not allowed here.
  Jul 15 13:16:16 ubuntuserver-1 collectd[14082]: supervised by systemd, will signal readyness
  Jul 15 13:16:16 ubuntuserver-1 collectd[14082]: Initialization complete, entering read-loop.


Thanks for spending time here, rpv-tomsk!

**Comment by rpv-tomsk** [comment]
Option names are case-sensitive.

**Comment by brianr-dk** [comment]
Thanks.

I thought so, and have actually tried with both wsrepstats, wsrepstatsWrongName as well as WsrepStats, and they all fail the same way when service collectd is restarted  :(  

Which is why I suspect that my exact version of that plugin might be different than the one at 
https://github.com/collectd/collectd/blob/master/src/mysql.c   - which shows the correct spelling in its line 202. 
Although I donøt understand how it should differ, because it's installed as a basic    "apt install collectd" 

Is there a way to verify which plugin:MySql version I am using, so I don't waste more of anyones time with an old version that's not supposed to understand option  WsrepStats anyway? 

My version (as far as I know to ask:  
root@ubuntuserver-1:~# dpkg -l|grep collectd 
ii  collectd                              5.5.1-1build2                              amd64        statistics collection and monitoring daemon
ii  collectd-core                         5.5.1-1build2                              amd64        statistics collection and monitoring daemon (core system)
) 

Thanks.

**Comment by rpv-tomsk** [comment]
ouch, really. 5.5.1 is old for your goals.

**Comment by brianr-dk** [comment]
Yes, thanks :) 
I can see things have grown, since someone created the repositories for my xenial Ubuntu. 
All collectd was so far simply installed with apt-get update/upgrade and install  collect/collectd-core, but that only gets us (xenial) users the 5.5.1 

I have now "manually" updated from  5.5.1   to   5.8  and now it actually accepts the Option WsrepStats (y) 
I just have to see now, that it can also submit metric values to the Graphite instance - so far only the "counters/names" have been published and can be selected in Grafana. But no real values has come across yet :) 

Thanks for answering!

**Comment by brianr-dk** [comment]
Closed because repos for Ubuntu 16.04 only contains collectd 5.5.1, which is prior to the invention of "Option wsrepstats". 
It perfectly explains why I can't use that option :D
